tests/scripts/run-pmi-diffs.py

```diff
@@ -0,0 +1,704 @@
+        if returncode != 0:
+            log('Return code = %s' % returncode)
+            result = 1
+
+    # Did we get any diffs?
+
+    baseOutputDir = os.path.join(asmRootPath, "base", "base")
+    if not testing and not os.path.isdir(baseOutputDir):
+        log("ERROR: base asm not generated")
+        return 1
+
+    # Do the jit-analyze comparison:
+    #   dotnet c:\gh\jitutils\bin\jit-analyze.dll --base f:\output\diffs\base\diff --recursive --diff f:\output\diffs\diff\diff
+
+    command = ["dotnet", jitAnalyzePath, "--base", baseOutputDir, "--diff", diffOutputDir]
+    log('Invoking: %s' % (' '.join(command)))
+    if not testing:
+        proc = subprocess.Popen(command, env=my_env)
+        output,error = proc.communicate()
+        returncode = proc.returncode
+        if returncode != 0:
+            log('Return code = %s' % returncode)
+            log('Compare: %s %s' % (baseOutputDir, diffOutputDir))
+
+    # Shutdown the dotnet build servers before cleaning things up
+    # TODO: make this shutdown happen anytime after we've run any 'dotnet' commands. I.e., try/finally style.
+
+    log('Shutting down build servers')
+    command = ["dotnet", "build-server", "shutdown"]
+    log('Invoking: %s' % (' '.join(command)))
+    if not testing:
+        proc = subprocess.Popen(command, env=my_env)
+        output,error = proc.communicate()
+        returncode = proc.returncode
+        if returncode != 0:
+            log('Return code = %s' % returncode)
+
+    return result
+
+##########################################################################
+# Main
+##########################################################################
+
+def main(args):
+
+    global arch, ci_arch, build_type, base_root, diff_root, scratch_root, skip_baseline_build, skip_diffs, target_branch, commit_hash
+    global my_env
+    global base_layout_root
+    global diff_layout_root
+    global baseCoreClrPath
+    global testing
+
+    arch, ci_arch, build_type, base_root, diff_root, scratch_root, skip_baseline_build, skip_diffs, target_branch, commit_hash = validate_args(args)
+
+    my_env = os.environ
+
+    if not testing and not os.path.isdir(diff_root):
+       log('ERROR: root directory for coreclr diff tree not found: %s' % diff_root)
+       return 1
+
+    # Check the diff layout directory before going too far.
+
+    diff_layout_root = os.path.join(diff_root,
+                                    'bin',
+                                    'tests',
+                                    '%s.%s.%s' % (clr_os, arch, build_type),
+                                    'Tests',
+                                    'Core_Root')
+
+    if not testing and not os.path.isdir(diff_layout_root):
+       log('ERROR: diff test overlay not found or is not a directory: %s' % diff_layout_root)
+       return 1
+
+    # Create the scratch root directory
+
+    if not testing:
+        try:
+            os.makedirs(scratch_root)
```

Comment:
This should also use tempfile see https://github.com/dotnet/coreclr/blob/master/tests/runtest.py#L448